aws/rds_cluster_snapshots_test.go

```diff
@@ -0,0 +1,169 @@
+	input := &rds.DescribeDBClustersInput{
+		DBClusterIdentifier: name,
+	}
+
+	for i := 0; i < 90; i++ {
+		_, err := svc.DescribeDBClusters(input)
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(1 * time.Second)
+		logging.Logger.Debug("Waiting for RDS DB Cluster to be created")
+	}
+
+	return RdsClusterAvailableError{name: *name}
+}
+
+// Built-in waiter function WaitUntilDBClusterSnapshotAvailable not working as expected.
+// Created a custom one
+func waitUntilRdsClusterSnapshotAvailable(svc *rds.RDS, clusterName *string, snapshotName *string) error {
+	input := &rds.DescribeDBClusterSnapshotsInput{
+		DBClusterIdentifier:         clusterName,
+		DBClusterSnapshotIdentifier: snapshotName,
+	}
+	for i := 0; i < 90; i++ {
+		_, err := svc.DescribeDBClusterSnapshots(input)
+		if err != nil {
+			return err
+		}
+
+		time.Sleep(1 * time.Second)
+		logging.Logger.Debug("Waiting for RDS DB Cluster Snapshot to be created")
+	}
+
+	return RdsClusterSnapshotAvailableError{clusterName: *clusterName, snapshotName: *snapshotName}
+}
+
+// createTestDBCluster generates a test DB Cluster since snapshots can only be created
+// from an existing DB Cluster
+func createTestDBCluster(t *testing.T, session *session.Session, name string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBClusterInput{
+		DBClusterIdentifier: awsgo.String(name),
+		Engine:              awsgo.String("aurora-mysql"),
+		MasterUsername:      awsgo.String("gruntwork"),
+		MasterUserPassword:  awsgo.String("password"),
+	}
+
+	_, err := svc.CreateDBCluster(input)
+	require.NoError(t, err)
+	waitUntilRdsClusterAvailable(svc, &name)
+
+}
+
+// createTestRDSClusterSnapshot generates a test DB Snapshot
+func createTestRDSClusterSnapshot(t *testing.T, session *session.Session, clusterName string, snapshotName string) {
+	svc := rds.New(session)
+	input := &rds.CreateDBClusterSnapshotInput{
+		DBClusterIdentifier:         awsgo.String(clusterName),
+		DBClusterSnapshotIdentifier: awsgo.String(snapshotName),
+	}
+
+	_, err := svc.CreateDBClusterSnapshot(input)
+	require.NoError(t, errors.WithStackTrace(err))
+	waitUntilRdsClusterSnapshotAvailable(svc, &clusterName, &snapshotName)
+
+	result, err := svc.DescribeDBClusterSnapshots(&rds.DescribeDBClusterSnapshotsInput{
+		DBClusterSnapshotIdentifier: &snapshotName,
+	})
+
+	// Generate a tag
+	var tags = []*rds.Tag{
+		{
+			Key:   awsgo.String("snapshot-tag1-key"),
+			Value: awsgo.String("snapshot-tag1-value"),
+		},
+	}
+
+	// Tag a DB Cluster snapshot resource
+	for _, database := range result.DBClusterSnapshots {
+		arn := database.DBClusterSnapshotArn
+		svc.AddTagsToResource(&rds.AddTagsToResourceInput{
+			ResourceName: awsgo.String(*arn),
+			Tags:         tags,
+		})
+	}
+
+}
+
+// TestFilterRDSClusterSnapshot tests for filtering cluster snapshots basing on matching include and exclude rules in the config file.
+func TestFilterRDSClusterSnapshot(t *testing.T) {
```

Comment:
How long does this test take to run?

Reply:
I am not sure. Is there a way to determine how long this test will run ?